libazureinit/src/provision/config.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+use crate::error::Error;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use toml;
+use tracing;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NetworkConfig {
+    pub manage_configuration: bool,
+    pub network_manager: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct SshConfig {
+    pub authorized_keys_path: String,
+    pub configure_password_authentication: bool,
+    pub authorized_keys_path_query_mode: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ImdsConfig {
+    pub connection_timeout_secs: f64,
+    pub read_timeout_secs: u32,
+    pub retry_timeout_secs: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ProvisioningMediaConfig {
+    pub enable: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AzureProxyAgentConfig {
+    pub enable: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WireserverConfig {
+    pub connection_timeout_secs: f64,
+    pub read_timeout_secs: u32,
+    pub retry_timeout_secs: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ErrorHandling {
+    pub strict_validation: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TelemetryConfig {
+    pub kvp_diagnostics: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Config {
```

Comment:
If you put `#[serde(default)]` (see https://serde.rs/container-attrs.html) on this struct, any missing fields will be filled in by the implementation of `Default` so users don't need to fill in a complete config file if they want to just change one section setting.